src/bootstrap/run.rs

```diff
@@ -275,3 +278,34 @@ impl Step for GenerateWindowsSys {
         builder.run(&mut cmd);
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct GenerateCompletions;
+
+impl Step for GenerateCompletions {
+    type Output = ();
+
+    /// Uses `clap_complete` to generate shell completions.
+    fn run(self, builder: &Builder<'_>) {
+        let [bash, fish, powershell] = ["x.sh", "x.fish", "x.ps1"]
+            .map(|filename| builder.src.join("src/etc/completions").join(filename));
+        if let Some(comp) = get_completion(shells::Bash, &bash) {
+            std::fs::write(&bash, comp).expect("writing bash completion");
+        }
+        if let Some(comp) = get_completion(shells::Fish, &fish) {
+            std::fs::write(&fish, comp).expect("writing fish completion");
+        }
+        if let Some(comp) = get_completion(shells::PowerShell, &powershell) {
+            std::fs::write(&powershell, comp).expect("writing powershell completion");
+        }
+        // FIXME: zsh generated completions don't seem to work well with subcommands
```

Comment:
Can we report this upstream to `clap_complete`?

Reply:
I'll look at properly diagnosing and reporting the issue today

Reply:
https://github.com/clap-rs/clap/issues/4898

Reply:
Even after fixing this error, flags don't seem to be being suggested after the subcommand and the path completion isn't working at all 🤔

Reply:
Merged now, I'll investigate the remaining issues with Zsh soon. I'll open a follow-up bumping the clap_complete version when I have a fix